[LsmnBmnc] iP

LsmnBmnc opened the pull request:
# Nana

> **"Your mind is for having ideas, not holding them."** — David Allen ([source](https://gettingthingsdone.com))

**DukePro** frees your mind from having to remember things you need to do. It's:

- **Text-based**
- **Easy to learn**
- **~~FAST~~ SUPER FAST** to use

## Getting Started

All you need to do is:

1. [Download it from here](https://example.com/download).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you :wink:

And it is **FREE**!

## Features

- [x] **Managing tasks** using `process()`
- **Managing deadlines** (coming soon)
- **Reminders** (coming soon)

## sample code
```java
public Task(String description) {
    this.description = description;
    this.isDone = false;
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

LsmnBmnc committed:
Level-2

LsmnBmnc committed:
Level-3

LsmnBmnc committed:
Level-4

LsmnBmnc committed:
A-TextUiTesting

LsmnBmnc committed:
Level-5

LsmnBmnc committed:
Level-6

LsmnBmnc committed:
Level-0

LsmnBmnc committed:
Level-1

LsmnBmnc committed:
Level-6

LsmnBmnc committed:
Level-6

LsmnBmnc committed:
Level-7

LsmnBmnc committed:
Branch-Level-8

LsmnBmnc committed:
Merge branch 'branch-Level-8'

LsmnBmnc committed:
More-OOP-1

LsmnBmnc committed:
More-OOP-2

LsmnBmnc committed:
A-MoreOOP

LsmnBmnc committed:
Merge branch 'More-OOP'

LsmnBmnc committed:
A-Packages

LsmnBmnc committed:
A-packages

LsmnBmnc committed:
Merge branch 'A-Packages'

LsmnBmnc committed:
Merge branch 'add-gradle-support'

LsmnBmnc committed:
A-gradle

LsmnBmnc committed:
A-JUnit

LsmnBmnc committed:
A-jar

LsmnBmnc committed:
Branch-Level-9

LsmnBmnc committed:
branch-A-CodingStandard

LsmnBmnc committed:
branch-Level-9

LsmnBmnc committed:
branch-Level-9

LsmnBmnc committed:
Merge branch 'branch-A-CodingStandard'

LsmnBmnc committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	data/Nana.txt

LsmnBmnc committed:
A-CheckStyle

LsmnBmnc committed:
Merge tag 'A-CheckStyle'

LsmnBmnc committed:
branch-Level-10

LsmnBmnc committed:
Merge branch 'branch-Level-10'

darHH commented on src/main/java/nana/logic/Deadline.java:
Overall very nice code, I like how simple yet clear your javadoc comments are.

darHH commented on src/main/java/nana/logic/Deadline.java:
```diff
@@ -0,0 +1,61 @@
+     */
+    public Deadline(String description, String by, boolean isDone) {
+        super(description, isDone);
+        this.byString = by;
```
I was abit confused upon reading byString and by, perhaps you could be more specific with byString to byDateString.

darHH commented on src/main/java/nana/logic/Event.java:
```diff
@@ -0,0 +1,61 @@
+     * Constructs a new Event instance with the specified description, start time, and end time.
+     *
+     * @param description the description of the event task
+     * @param startTime the start time of the event task
+     * @param endTime the end time of the event task
```
Perhaps you could include in HH:mm format in javadoc comments (like what you did for deadline class)

darHH commented on src/main/java/nana/logic/Parser.java:
Perhaps you could add javadoc commetns for the methods in this class.

darHH commented on src/main/java/nana/logic/Ui.java:
Perhaps to prevent repeating the printing the line (many dashes), use class member field like "private static final String BORDER = "    ____________________________________________________________";"

darHH reviewed (commented):
overall very nicely written code, just a few things i noticed. Good job!

delonlee01 commented on src/main/java/nana/logic/Deadline.java:
```diff
@@ -0,0 +1,61 @@
+     * @return a string representation of the deadline task for storage
+     */
+    @Override
+    public String toStorage() {
```
Maybe can consider `toStorageString`?

delonlee01 commented on src/main/java/nana/logic/Storage.java:
```diff
@@ -0,0 +1,84 @@
+ */
+public class Storage {
+
+    private String filepath;
```
Shouldn't this be written in camelCase?

delonlee01 commented on src/main/java/nana/logic/Storage.java:
```diff
@@ -0,0 +1,84 @@
+    /**
+     * Updates the task file with the current list of tasks.
+     */
+    public static void updateTxt() {
```
Maybe you can consider `updateTextFile`?

delonlee01 commented on src/main/java/nana/logic/Storage.java:
```diff
@@ -0,0 +1,84 @@
+    /**
+     * Deletes the task file.
+     */
+    public static void deleteTxt() {
```
Similarly, maybe you can consider `deleteTextFile`?

delonlee01 commented on src/main/java/nana/logic/TaskList.java:
```diff
@@ -0,0 +1,195 @@
+    public String addTodo(ArrayList<String> parsed) {
+        tasks.add(new Todo(parsed.get(0)));
+        addCount();
+        String s = Ui.printAddTodo(tasks.get(taskCount - 1), taskCount);
```
Maybe this variable name can be more intuitive?

delonlee01 commented on src/main/java/nana/logic/TaskList.java:
```diff
@@ -0,0 +1,195 @@
+        taskCount++;
+    }
+
+    private void minsCount() {
```
Should this be `minusCount`? Maybe you can consider `decrementCount` also

delonlee01 reviewed (commented):
Great progress! 👍🏻

LsmnBmnc committed:
A-Assertions

LsmnBmnc committed:
A-CodeQuality

LsmnBmnc committed:
Merge pull request #1 from LsmnBmnc/branch-A-Assertions

A-Assertions

LsmnBmnc committed:
Merge branch 'master' into branch-A-CodeQuality

LsmnBmnc committed:
Merge branch 'branch-A-CodeQuality'

LsmnBmnc committed:
BCD-Extension

LsmnBmnc committed:
Merge branch 'branch-BCD-Extension'

LsmnBmnc committed:
Add more javadoc

LsmnBmnc committed:
Add more scenarios to be handled by exception

LsmnBmnc committed:
Add an assertion for process

LsmnBmnc committed:
Remove meaningless javadocs

LsmnBmnc committed:
Add some javadocs

LsmnBmnc committed:
Remove unused method

The method parseAddTask is removed because it is unused.

Deleting this improves code quality.

LsmnBmnc committed:
Add User guide

LsmnBmnc committed:
Add Ui.png

LsmnBmnc committed:
Update Ui.png path

LsmnBmnc committed:
Add jar configuration

LsmnBmnc committed:
Update README.md

LsmnBmnc committed:
Fix bug for Gui button
